src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,677 @@
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="long"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        public bool TryGetInt64(out long value) => long.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="float"/> equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="float"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        /// <remarks> 
+        ///   On .NET Core this method does not return <see langword="false"/> for values larger than 
+        ///   <see cref="float.MaxValue"/> (or smaller than <see cref="float.MinValue"/>), 
+        ///   instead <see langword="true"/> is returned and <see cref="float.PositiveInfinity"/> (or 
+        ///   <see cref="float.NegativeInfinity"/>) is emitted. 
+        /// </remarks> 
+        public bool TryGetSingle(out float value) => float.TryParse(_value, NumberStyles.Float, CultureInfo.InvariantCulture, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="double"/> equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="double"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        /// <remarks> 
+        ///   On .NET Core this method does not return <see langword="false"/> for values larger than 
+        ///   <see cref="float.MaxValue"/> (or smaller than <see cref="float.MinValue"/>), 
+        ///   instead <see langword="true"/> is returned and <see cref="float.PositiveInfinity"/> (or 
+        ///   <see cref="float.NegativeInfinity"/>) is emitted. 
+        /// </remarks> 
+        public bool TryGetDouble(out double value) => double.TryParse(_value, NumberStyles.Float, CultureInfo.InvariantCulture, out value);
```

Comment:
nit: Specifying NumberStyles is unnecessary (since the default Parse uses that NumberStyles.Float anyway). Consider using the other overload (same elsewhere):
```suggestion
        public bool TryGetDouble(out double value) => double.TryParse(_value, CultureInfo.InvariantCulture, out value);
```

Reply:
I planned to, but `double.TryParse` seems to have only overloads with `IFormatProvider`, which takes `NumerStyles` as well: https://docs.microsoft.com/en-us/dotnet/api/system.double.tryparse?view=netcore-3.0 (`double.Parse` have it though).

Reply:
Ah... Good point. I was looking at the `Parse` APIs and assumed TryParse would have similar overloads. That is an odd inconsistency. It's fine to leave it as is then.

Reply:
@stephentoub, though we can't fix this down-level, do you think its worth adding an issue to make the API overloads consistent on Parse/TryParse (for all the numeric types - decimal, float, double, etc.)?

Reply:
@kasiabulat, when specifying the NumberStyle for the Try* APIs, make sure to match the ones used by default for Double/Float (include `NumberStyles.AllowThousands`).

https://github.com/dotnet/corefx/blob/bf8945497724dbe7c69ce8aaf4a24ac4fffc1a72/src/Common/src/CoreLib/System/Double.cs#L299-L303

Reply:
> do you think its worth adding an issue to make the API overloads consistent on Parse/TryParse (for all the numeric types - decimal, float, double, etc.)?

What is the inconsistency?

Reply:
The `Parse` overloads let you pass in the `NumberStyles`, and `IFormatProvider` independently. So, the caller can call the overload with fewer arguments if they want one of the arguments to remain default.
The `TryParse` overloads force you to pass in a `NumberStyles` if all you want to change is `IFormatProvider` (and vice versa).

Parse overloads:
https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Runtime/ref/System.Runtime.cs#L2201-L2205
TryParse overloads:
https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Runtime/ref/System.Runtime.cs#L2226-L2229

Reply:
Why would we allow thousands (e.g. commas)?  IIRC JSON doesn't allow commas, so we should never been in a state where we needed to parse them.  We should use the overloads of non-Try Parse that aren't flexible, instead of making the Try be as flexible in a way that shouldn't matter.

Reply:
> so we should never been in a state where we needed to parse them

That's a good point. Even in the ctor that accepts string, we detect/reject numbers that aren't JSON compliant.

> We should use the overloads of non-Try Parse that aren't flexible, instead of making the Try be as flexible in a way that shouldn't matter.

The question is what NumberStyle/parameter fits best to only allow parsing of input that is valid JSON. Do we want `JsonNumber.TryGetDecimal` to return true for "1e1", even if `decimal.TryParse("1e1", out _)` returns false? If so, leaving it as `NumberStyles.Float` makes sense.

Reply:
Is the final conclusion to keep `NumberStyles.Float` everywhere (as it is now, for float, double and decimal, in Get and TryGet)?

Reply:
Yep. The only other difference is trailing sign, which isn't valid in JSON either, so we should leave it as is (i.e. use `NumberStyles.Float`) for all three.
```C#
decimal.Parse("1000-"); // valid here, but not valid JSON number
```